resolver/type_syntax.h

```diff
@@ -47,20 +47,49 @@ struct ParsedSig {
     const TypeArgSpec &findTypeArgByName(core::NameRef name) const;
 };
 
+struct TypeSyntaxArgs {
+    bool allowSelfType = false;
+    bool allowRebind = false;
+    core::SymbolRef untypedBlame;
+
+    TypeSyntaxArgs(const bool allowSelfType, const bool allowRebind, core::SymbolRef untypedBlame)
+        : allowSelfType(allowSelfType), allowRebind(allowRebind), untypedBlame(untypedBlame) {}
+
+    static TypeSyntaxArgs forGetResultType(core::SymbolRef untypedBlame) {
+        return TypeSyntaxArgs(true, false, untypedBlame);
+    }
+
+    static TypeSyntaxArgs forParseSig(core::SymbolRef untypedBlame) {
+        return TypeSyntaxArgs(true, false, untypedBlame);
+    }
+
+    TypeSyntaxArgs noRebind() const {
+        return TypeSyntaxArgs(allowSelfType, false, untypedBlame);
+    }
```

Comment:
What do you think about naming these all `with` and `without`? Instead of `with` and `no`?

Reply:
Sounds good to me, I'll make that change.